File: server/test-api.js
const axios = require('axios');

const API_BASE_URL = 'http://localhost:5000';

async function testAPI() {
  console.log('🧪 Testing Notecraft Pro API...\n');

  try {
    // Test health endpoint
    console.log('1. Testing health endpoint...');
    const healthResponse = await axios.get(`${API_BASE_URL}/api/health`);
    console.log('✅ Health check passed:', healthResponse.data);
    console.log('');

    // Test humanize endpoint
    console.log('2. Testing humanize endpoint...');
    const testText = "The implementation of artificial intelligence in modern business environments has demonstrated significant potential for enhancing operational efficiency and productivity metrics across various organizational structures.";
    
    const humanizeResponse = await axios.post(`${API_BASE_URL}/api/humanize`, {
      text: testText,
      tone: 'professional',
      style: 'business',
      length: 'maintain'
    });
    
    console.log('✅ Humanize test passed!');
    console.log('Original:', testText);
    console.log('Humanized:', humanizeResponse.data.humanizedText);
    console.log('Word count:', humanizeResponse.data.wordCount);
    console.log('');

    // Test AI detection endpoint
    console.log('3. Testing AI detection endpoint...');
    const detectResponse = await axios.post(`${API_BASE_URL}/api/detect`, {
      text: humanizeResponse.data.humanizedText
    });
    
    console.log('✅ AI detection test passed!');
    console.log('AI Score:', detectResponse.data.aiScore + '%');
    console.log('Human Score:', detectResponse.data.humanScore + '%');
    console.log('Assessment:', detectResponse.data.assessment);
    console.log('');

    // Test advanced humanization
    console.log('4. Testing advanced humanization...');
    const advancedResponse = await axios.post(`${API_BASE_URL}/api/humanize-advanced`, {
      text: testText,
      tone: 'friendly',
      style: 'casual',
      targetAudience: 'general',
      length: 'maintain',
      creativity: 'balanced'
    });
    
    console.log('✅ Advanced humanization test passed!');
    console.log('Advanced humanized:', advancedResponse.data.humanizedText);
    console.log('Settings used:', advancedResponse.data.settings);
    console.log('');

    console.log('🎉 All API tests passed successfully!');
    console.log('🚀 Your Notecraft Pro API is ready for production!');

  } catch (error) {
    console.error('❌ API test failed:', error.response?.data || error.message);
    console.log('');
    console.log('Troubleshooting tips:');
    console.log('1. Make sure the server is running: npm start');
    console.log('2. Check that OPENAI_API_KEY is set in your .env file');
    console.log('3. Verify the API_BASE_URL is correct');
  }
}

// Run the test
testAPI(); 

File: api/test_api.py
#!/usr/bin/env python3
"""
Test script for the FastAPI humanizer API
"""
import requests
import json
import sys

def test_api():
    """Test the FastAPI humanizer endpoint"""
    base_url = "http://localhost:8000"
    
    # Test health endpoint
    print("🔍 Testing health endpoint...")
    try:
        response = requests.get(f"{base_url}/healthz")
        if response.status_code == 200:
            print("✅ Health check passed!")
            print(f"Response: {response.json()}")
        else:
            print(f"❌ Health check failed: {response.status_code}")
            return False
    except requests.RequestException as e:
        print(f"❌ Error connecting to server: {e}")
        print("Make sure the FastAPI server is running with: python main.py")
        return False
    
    # Test humanize endpoint
    print("\n🤖 Testing humanize endpoint...")
    test_text = "This is an AI-generated text that needs humanization. It might sound robotic and needs to be more natural."
    
    payload = {
        "text": test_text,
        "tone": "friendly",
        "style": "casual",
        "length": "maintain"
    }
    
    try:
        response = requests.post(
            f"{base_url}/humanize",
            json=payload,
            headers={"Content-Type": "application/json"}
        )
        
        if response.status_code == 200:
            result = response.json()
            print("✅ Humanization successful!")
            print(f"Original: {result['originalText']}")
            print(f"Humanized: {result['humanizedText']}")
            print(f"Word count: {result['wordCount']}")
            print(f"Character count: {result['characterCount']}")
        else:
            print(f"❌ Humanization failed: {response.status_code}")
            print(f"Error: {response.text}")
            return False
            
    except requests.RequestException as e:
        print(f"❌ Error during humanization: {e}")
        return False
    
    return True

if __name__ == "__main__":
    print("🚀 Starting FastAPI Humanizer API Test")
    print("=" * 50)
    
    success = test_api()
    
    if success:
        print("\n🎉 All tests passed! The FastAPI server is working correctly.")
    else:
        print("\n💥 Tests failed. Please check the server and try again.")
        sys.exit(1)